lib/pmdtester/builders/diff_report_builder.rb

```diff
@@ -210,24 +213,38 @@ def build_errors_table_head(doc)
 
     def build_errors_table_body(doc, errors)
       doc.tbody do
-        b_index = 1
-        errors.each do |pmd_error|
-          doc.tr(class: pmd_error.branch == 'base' ? 'b' : 'a') do
-            # The anchor
-            doc.td do
-              doc.a(id: "B#{b_index}", href: "#B#{b_index}") { doc.text '#' }
-            end
+        Array(0..(errors.size - 1)).each do |i|
+          build_errors_table_row(doc, i, errors)
+        end
+      end
+    end
 
-            # The error message
-            doc.td pmd_error.msg
+    def build_errors_table_row(doc, i, errors)
```

Comment:
I would write this different - since the method does two things: either it outputs a diff if we have exactly to errors, one from base and one from patch. Otherwise it renders error by error.

I suggest to create two methods for that.
You can then call the methods in `build_errors_table_body` this way:

```
if ReportDiff.comparable?(errors)
    build_errors_table_row_diff(doc, errors[0], errors[1])
else
    errors.each do |pmd_error|
        build_error_table_row(doc, pmd_error)
    end
end
```

What do you think?

Reply:
Maybe we should do it in `build_errors_table_row` since there are duplicate logic in `build_errors_table_row_diff` and `build_error_table_row`.